df.viewscreen.xml

```diff
@@ -1730,55 +1726,59 @@
             <int32_t name='belief_strength' since='v0.40.01'/>
 
-            <compound name="unk_324">
+            <compound name='appearance'>
                 <int32_t name='unk_1'/>
-                <int16_t name='unk_2'/>
+                <int16_t name='caste_index' refers-to='$global.world.raws.creatures.list_creature[$]' comment='also refers to $global.world.raws.creatures.list_caste[$]'/>
                 <int16_t name='unk_3'/>
                 <int32_t name='unk_4'/>
-                <static-array name='unk_5' count='6' type-name='unit_attribute'/>
+                <static-array name='physical_attributes' count='6' type-name='unit_attribute' index-enum='physical_attribute_type'/>
                 <int32_t name='unk_6'/>
-                below vectors (unk_7-unk_13) populated on PhysicalTraits page; length depends on caste, but contents are consistent with same caste and save
-                <stl-vector name='unk_7' type-name='int32_t'/>
-                <stl-vector name='unk_8' type-name='int32_t'/>
+                below vectors populated on PhysicalTraits page; lengths depend on caste
+                <stl-vector name='body_modifiers' type-name='int32_t' index-refers-to='(find-creature $$._parent.race).caste[$$._parent.caste].body_appearance_modifier[$]'/>
+                <stl-vector name='bp_modifiers' type-name='int32_t' index-refers-to='(find-creature $$._parent.race).caste[$$._parent.caste].bp_appearance.modifiers[$]'/>
                 <int32_t name='unk_9'/>
-                <stl-vector name='unk_10' type-name='int32_t'/>
-                <stl-vector name='unk_11' type-name='int32_t'/>
-                <stl-vector name='unk_12' type-name='int32_t'/>
-                <stl-vector name='unk_13' type-name='int32_t'/>
-                <stl-vector name='unk_14' type-name='int16_t'/> physical traits? changed by randomizing in PhysicalTraits page
-
-                <static-array name='unk_15' count='2'>
-                    <compound>
-                        <pointer name='unk_15_a'/>
-                        <int16_t name='unk_15_b'/>
-                    </compound>
-                </static-array>
-                <stl-vector name='unk_16' type-name='int32_t'/>
+                <stl-vector name='tissue_style' type-name='int32_t'/>
+                <stl-vector name='tissue_style_civ_id' type-name='int32_t'/>
+                <stl-vector name='tissue_style_id' type-name='int32_t'/>
+                <stl-vector name='tissue_style_type' type-name='int32_t'/>
+                <stl-vector name='tissue_length' type-name='int32_t'/>
+
+                <pointer name='unk_15_a'/>
+                <int16_t name='appearance_gene_count'/>
+                <pointer name='unk_15_b'/>
+                <int16_t name='color_gene_count'/>
+
+                <stl-vector name='color_modifiers' type-name='int32_t' index-refers-to='(find-creature $$._parent.race).caste[$$._parent.caste].color_modifiers[$]/>
```

Comment:
There's a missing quote at the end of this line, by the way. It looks like I forgot to mention it earlier, but I had to fix it to test this, so no need to change it here.